scrc/dataset_creation/dataset_creator.py

```diff
@@ -266,20 +270,100 @@ def get_dataset_folder(self):
+
         self.logger.info(f"Creating {self.dataset_name} dataset")
 
         # TODO in the future: maybe save text as list of paragraphs
         # TODO make sure that the same data is saved to kaggle, csv and huggingface format!
 
-        dataset, labels = self.prepare_dataset(save_reports)
-        dataset = dataset.shuffle(seed=42)
-        self.save_dataset(dataset, labels, self.get_dataset_folder(), self.split_type,
-                          sub_datasets=sub_datasets, kaggle=kaggle, save_reports=save_reports)
+        not_created = []
+        created = []
+
+        datasets_list = []
+        labels_list = []
+
+        for court_string in court_list:
+            self.logger.info(f"Creating dataset for {court_string}")
+            dataset, labels = self.prepare_dataset(save_reports, court_string=court_string)
+
+            if len(dataset) == 0:
+                self.logger.info(f"Dataset for {court_string} could not be created")
+                not_created.append(court_string)
+            else:
+                dataset = dataset.shuffle(seed=42)
+                if concatenate:
+                    datasets_list.append(dataset)
+                    labels_list.append(labels)
+                else:  # save each dataset separately
+                    save_path = Path(os.path.join(self.get_dataset_folder(), court_string))
+                    self.save_dataset(dataset, labels, save_path, self.split_type,
+                                      sub_datasets=sub_datasets, kaggle=kaggle, save_reports=save_reports)
+
+                created.append(court_string)
+            self.logger.info(f"Empty courts: {not_created}")
+            self.logger.info(f"Created courts: {created}")
+
+        if concatenate:
+            dataset = concatenate_datasets(datasets_list)
+            labels = labels_list[0]
+
+            # check if export folder already exists and increment the name index if it does
+            counter = 1
+            while os.path.exists(f"{self.get_dataset_folder()}/concat_{counter}"):
+                counter += 1
+            export_path = Path(f"{self.get_dataset_folder()}/concat_{counter}")
+
+            self.save_dataset(dataset, labels, export_path, "all_train", kaggle=kaggle,
+                              save_reports=save_reports)
+
+        self.logger.info(f"{len(not_created)} courts not created (since it was empty): {not_created}")
+        self.logger.info(f"{len(created)} courts created: {created}")
+
+    # returns all courts that can be generated without any problems based on the latest state of knowledge
+    def get_court_list(self):
+        """
+        :return: list
+        """
+        # get names of all courts
+        court_list_tmp = get_all_courts()
+
+        # taking all folder names from /data/datasets as a list to know which courts are already generated
+        courts_done = os.listdir(os.path.join(ROOT_DIR, str(self.datasets_subdir)))
```

Comment:
the path.join with ROOT_DIR is not necessary here, I think

Reply:
Oh yes, you're right.